pinot-core/src/main/java/org/apache/pinot/core/startree/StarTreeUtils.java

```diff
@@ -363,8 +372,14 @@ public static BaseProjectOperator<?> createStarTreeBasedProjectOperator(IndexSeg
     ExpressionContext[] groupByExpressions =
         queryContext.getGroupByExpressions() != null ? queryContext.getGroupByExpressions()
             .toArray(new ExpressionContext[0]) : null;
+
+    List<Pair<AggregationFunction, AggregationFunctionColumnPair>> aggregations =
```

Comment:
Avoid using functional API at query path as they are way more expensive

Reply:
It probably doesn't matter here since this is once per segment call. This is one of the best practices for engine code though as I've found functional API is usually 2-3 times slower than regular copy, besides the higher memory footprint